### DISCUSSION TITLE
Create a SemanticSegmentationRandomWindowGeoDataset intersecting images and labels uri.

### DISCUSSION THREAD

**Discussion by lcoandrade** [azavea/raster-vision#1808] [Q&A] [answered]
Hi there!
I'm wondering if I can make a SemanticSegmentationRandomWindowGeoDataset from the intersection of images and labels.
I'm trying to use `aoi_uri` with a geojson file with the boundary of image but the batch in the visualization seems not to be spatially paired.

This is how the data is spatially related.
<img width="763" alt="Captura de Tela 2023-06-21 às 23 03 09" src="https://github.com/azavea/raster-vision/assets/8769816/6846bf9c-d430-4bbe-97c1-ec48d2211935">

And this is what I get:
<img width="610" alt="Captura de Tela 2023-06-21 às 23 05 13" src="https://github.com/azavea/raster-vision/assets/8769816/3e4f0a63-bddf-458c-b2ad-438fdd4e9624">

I working on Kaggle with [this notebook](https://www.kaggle.com/code/luizclaudioandrade/rastervision-101).
The dataset is [this one](https://www.kaggle.com/datasets/luizclaudioandrade/naip-chesapeake-sample).

Can someone provide me some hints? What am I missing here?

Thanks!

**Comment by lcoandrade** [comment]
I tried cropping the images to stay like this:

Cropped image:
<img width="610" alt="Captura de Tela 2023-06-22 às 11 04 01" src="https://github.com/azavea/raster-vision/assets/8769816/8fb4bed3-2c44-4886-aef3-f217a0559bf9">

Cropped label:
<img width="610" alt="Captura de Tela 2023-06-22 às 11 04 13" src="https://github.com/azavea/raster-vision/assets/8769816/cecaa88b-507f-4dc5-a66a-5823dfa99d94">

But even though  I try to use image and label uri to theses files I get a batch that is not spatially related:
<img width="610" alt="Captura de Tela 2023-06-22 às 11 06 58" src="https://github.com/azavea/raster-vision/assets/8769816/7c32c27d-2663-4a17-81b3-4eeaa991f85f">

I tried using the piece of code:
```
data_augmentation_transform = A.Compose([
    A.Flip(),
    A.ShiftScaleRotate(),
    A.OneOf([
        A.RandomBrightness(),
        A.RandomGamma(),
    ]),
    A.CoarseDropout(max_height=32, max_width=32, max_holes=5)
])

datasets = []
for train_image_uri in train_image_uris:
    dataset = SemanticSegmentationRandomWindowGeoDataset.from_uris(
        class_config=class_config,
        image_uri=train_image_uri,
        label_raster_uri=train_label_uri,
        size_lims=(256, 512),
        out_size=256,
        max_windows=400,
        #transform=data_augmentation_transform,
        #aoi_uri=aoi_uri,
    )
    datasets.append(dataset)

concat_dataset = ConcatDataset(datasets)

len(concat_dataset)
```

I'm out of ideas...

**Comment by lcoandrade** [comment]
Maybe the problem is related to the pixels sizes.
Pixel sizes:
naip = 0.6m
chesapeake = 1m

I was thinking the the random windows were geo...

I'll try again warping the Chesapeake to have the same pixel size as naip.

**Comment by AdeelH** [comment]
Yeah, this is a bug in v0.20 that should be fixed on the master branch (#1740). The bug manifests when the image and label TIFFs don't have the same extent. Can you try installing Raster Vision from the current master branch?

The following works on Kaggle for installing RV from the current master branch:
```
!git clone https://github.com/azavea/raster-vision.git

!pip install -e raster-vision/rastervision_pipeline/
!rm -rf /opt/conda/lib/python3.10/site-packages/typing_extensions-4.5.0.dist-info
!pip install -e raster-vision/rastervision_aws_s3/
!pip install -e raster-vision/rastervision_aws_batch/
!pip install -e raster-vision/rastervision_core/
!rm -rf /opt/conda/lib/python3.10/site-packages/numpy-1.24.3.dist-info
!pip install -e raster-vision/rastervision_pytorch_learner/
!pip install -e raster-vision/rastervision_pytorch_backend/
!pip install -e raster-vision/
```
(There's some weirdness with pip and some packages which is why I had to add `rm -rf` lines.)

**Reply by lcoandrade** [reply]
I just tried running with the master branch.

When importing raster vision:
```
from rastervision.core.data import ClassConfig
```

I got this:
```
---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
Cell In[5], line 1
----> 1 from rastervision.core.data import ClassConfig
      3 class_config = ClassConfig(
      4     names=['water', 'wetlands', 'tree_canopy (tc)', 'shrubland',
      5           'low_veg', 'barren', 'structures', 'imp_surf',
      6           'imp_roads', 'tc_struct',
      7            'tc_imp_surf', 'tc_imp_roads'],
      8 )

ModuleNotFoundError: No module named 'rastervision'
```

How to address this?

**Reply by AdeelH** [reply]
Restart the kernel after installing. Just restart, not reset the whole session.

**Reply by lcoandrade** [reply, marked as answer]
Thank you! For now, I'm warping images and labels to have the same extent and pixel size. This solves the problem. Thanks for all the support!
